runtime/server/batch_query.go

```diff
@@ -0,0 +1,177 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/server/auth"
+)
+
+func (s *Server) QueryBatch(req *runtimev1.QueryBatchRequest, srv runtimev1.QueryService_QueryBatchServer) error {
+	ctx := srv.Context()
+
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.ReadOLAP) {
+		return ErrForbidden
+	}
+	var wg sync.WaitGroup
+
+	for _, query := range req.Queries {
+		wg.Add(1)
+		go func(query *runtimev1.QueryBatchSingleRequest) {
```

Comment:
I feel like I more often see `query := query` on the line before (instead of passing it as an arg), but both should work